### ISSUE TITLE
examples: run as root or change ownership

### ISSUE THREAD

**Issue by pohly** [intel/pmem-csi#926]
The pmem-app.yaml files run the PMEM-CSI container with the `pmem-csi` user. That makes the /data directory in the example unusable.

We either must change the ownership of it or change the user.

**Comment by avalluri** [comment]
@pohly As it is an example application, what about using other than pmem-csi container, say busybox?

**Comment by pohly** [comment]
That container then wouldn't have our fsdax check binary installed.

**Comment by avalluri** [comment]
> That container then wouldn't have our fsdax check binary installed.

Yes, but at least in our example, we do not mention using that binary. We mentioned running `mount` and `df` on the /data directory. As this is an example application I prefer we better use some non-pmem-csi image.

**Comment by pohly** [comment]
The ascii animation mentions it.

**Comment by pohly** [comment]
How is the busybox image better than our own?

**Comment by avalluri** [comment]
> The ascii animation mentions it.



> The ascii animation mentions it.

Okay, got it.

**Comment by avalluri** [comment]
> How is the busybox image better than our own?

It is not about better, I felt like it is an example app better use some generic not-pmem-csi specific image to avoid any confusion for the user.

**Comment by pohly** [comment]
The pod name should make that clear. I am not even sure whether users will notice what image it is based on.

**Comment by pohly** [comment]
> The ascii animation mentions it.

I don't want to make this issue any larger than it currently is, but we probably should rethink how we document the examples:
- pull them out of the https://intel.github.io/pmem-csi/0.9/docs/install.html#install-from-source section, using URLs instead of relative file paths
- document that the example apps are based on Debian and thus additional software can be installed with "apt update && apt get"
- mention pmem-dax-check

**Comment by avalluri** [comment]
> I don't want to make this issue any larger than it currently is,

We could make the examples work as we currently documented by setting the setuid bit of the `pmem-dax-check` binary as in #937.

**Comment by pohly** [comment]
I think the cleanest solution is to change permissions on the /data mount. That mirrors best practices. It can be done like this:
```
diff --git a/deploy/common/pmem-app-generic-ephemeral.yaml b/deploy/common/pmem-app-generic-ephemeral.yaml
index 0eeccf9c7..d4aa4fa98 100644
--- a/deploy/common/pmem-app-generic-ephemeral.yaml
+++ b/deploy/common/pmem-app-generic-ephemeral.yaml
@@ -6,6 +6,8 @@ apiVersion: v1
 metadata:
   name: my-csi-app-inline-volume
 spec:
+  securityContext:
+    fsGroup: 1000
   containers:
     - name: my-frontend
       image: intel/pmem-csi-driver-test:canary
pohly@pohly-desktop:/nvme/gopath/src/github.com/intel/pmem-csi$ kubectl exec -ti my-csi-app-inline-volume -- /bin/bash
pmem-csi@my-csi-app-inline-volume:/$ ls -l /data
total 16
drwxrws---. 2 root pmem-csi 16384 Apr 16 06:29 lost+found
pmem-csi@my-csi-app-inline-volume:/$ pmem-dax-check /data/test-file
/data/test-file: supports MAP_SYNC, content is zeroed: true

```

**Comment by pohly** [comment]
A short comment about the security context would be good, too:
```
# The container runs apps as user/group 1000/1000.
# This security context causes permissions of volume mounts
# to be adapted accordingly, see
# https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod
```

**Comment by avalluri** [comment]
> I think the cleanest solution is to change permissions on the /data mount. That mirrors best practices. It can be done like this:
> 
> ```
> diff --git a/deploy/common/pmem-app-generic-ephemeral.yaml b/deploy/common/pmem-app-generic-ephemeral.yaml
> index 0eeccf9c7..d4aa4fa98 100644
> --- a/deploy/common/pmem-app-generic-ephemeral.yaml
> +++ b/deploy/common/pmem-app-generic-ephemeral.yaml
> @@ -6,6 +6,8 @@ apiVersion: v1
>  metadata:
>    name: my-csi-app-inline-volume
>  spec:
> +  securityContext:
> +    fsGroup: 1000
> ```

I have initially tried with this solution, but the issue is that `pmem-dax-check` in our animation demo, tries to access `/home/test-file` to demonstrate that the non-pmem backed volume does not support MAP_SYNC. I couldn't figure out how to make make that work with the 'securityContext' solution:

```
$ kubectl exec my-csi-app-1 -- ls -dl /data
drwxrwsr-x. 3 root pmem-csi 4096 Apr 16 07:20 /data

$ kubectl exec my-csi-app-1 -- pmem-dax-check /data/test.txt
/data/test.txt: supports MAP_SYNC, content is zeroed: true

$ kubectl exec my-csi-app-1 -- ls -dl /home
drwxr-xr-x. 2 root root 4096 Mar 19 23:44 /home

$ kubectl exec my-csi-app-1 -- pmem-dax-check /home/test.txt
open /home/test.txt: permission denied
command terminated with exit code 2
```

Do you have any idea how to make it work even with pre-existed directories like `/home`.

**Comment by pohly** [comment]
> tries to access /home/test-file

The demo should use `/tmp/test-file` instead of that.

I already told Eric that the animation must be updated for 0.9. He is currently waiting for this issue here to be resolved and a new 0.9.1 release with the fix backported.
